Update with request module

MarkPaxton opened the pull request:
This version now handles digest requests as a wrapper around node request module. I have also added some tests, though of course you need an appropriate server to talk to.

Commit:
Starting to replace http with request library - bump version and add new
contributor

Commit:
Starting to replace http with request library - bump version and add new
contributor

Commit:
Update to use node request module to handle digest requests properly and
call callbacks compatible with request modile. Add grunt test setup.

MarkPaxton committed:
Merge branch 'master' of github.com:MarkPaxton/node-http-digest-client

Conflicts:
	http-digest-client.js
	package.json

MarkPaxton committed:
Merge with github version

kredmer committed:
added path parsing from given url

kredmer committed:
path parsing bugfix

kredmer committed:
ensure no basic auth with request module

simme commented:
- There's no new line at end of file in `.gitignore`.
- There's a project file committed.
- Why was the file moved?

MarkPaxton committed:
Added newline to .gitignore, removed and ignored .project file

MarkPaxton committed:
Merge pull request #1 from kredmer/master

Minor bugfix with new request module

MarkPaxton commented:
Hi, I have removed the project file and updated the ignore file. I have also merged in another bugfix.

Not sure which file you are referring to as moved though; do let me know.

Thanks
